docs/reference/api/storage/LittleFileSystem.md

```diff
@@ -1,11 +1,130 @@
 ## LittleFileSystem
 
-[Add description here.]
+The LittleFileSystem (littlefs) is a little fail-safe file system designed for embedded systems, specifically for microcontrollers that use flash storage.
+
+```
+   | | |     .---._____
+  .-----.   |          |
+--|o    |---| littlefs |
+--|     |---|          |
+  '-----'   '----------'
+   | | |
+```
+
+Microcontrollers and flash storage present three challenges for embedded storage: [power loss](#power-loss-resilience), [wear](#wear-leveling) and [limited RAM and ROM](#bounded-ram-and-rom). The LittleFileSystem provides a solution to all three of these problems.
+
+### Power loss resilience
+
+Embedded systems are usually without a shutdown routine and the notable lack of a user interface for recovery. With a file system that is not resilient to power loss, you rely on luck to not end up with a corrupted file system. The combination of persistent storage and unpredictable power loss creates bugs that are difficult to notice and that ruin the experience of unlucky users.
+
+The LittleFileSystem has a structure that is resilient to power loss. It uses checksums to limit the assumptions of how the physical storage reacts to power loss. The LittleFileSystem provides copy-on-write guarantees and keeps the storage on disk in a valid state.
+
+### Wear leveling
+
+Flash storage presents its own challenge: wear. Flash is a destructive form of storage, and continuously rewriting data to a block causes that block to wear out and become unwritable. File systems that don't account for wear can quickly burn through the blocks that store frequently updated metadata and result in the premature death of the system.
+
+We accounted for wear when we built the LittleFileSyste, and the underlying structure of the file system reactively mutates as the underlying storage develops errors throughout its lifetime. This results in a form of dynamic wear-leveling that extends the lifetime of the physical storage proportionally to the size of storage. With the LittleFileSystem, you can extend the lifetime of storage by increasing the size of storage, which is cheaper than upgrading the erase cycles on storage.
```

Comment:
I think its better to say we accounted for wear in the initial design phase itself instead of saying "when we built".

Reply:
Amanda and I have a rewrite in the notes Amanda added over here:
https://github.com/geky/mbed_Handbook/commit/d1406f10e5c68fc1f8164a783f30bec93dba0d46